boards/xtensa/nxp_adsp_rt595/nxp_adsp_rt595_defconfig

```diff
@@ -0,0 +1,7 @@
+CONFIG_SOC_SERIES_NXP_RT5XX=y
+CONFIG_SOC_NXP_RT595=y
+CONFIG_BOARD_NXP_ADSP_RT595=y
+
+CONFIG_GEN_ISR_TABLES=y
+CONFIG_GEN_IRQ_VECTOR_TABLE=n
+CONFIG_XTENSA_SMALL_VECTOR_TABLE_ENTRY=y
```

Comment:
Aren't all of these configs SoC quirks? If so we could probably move them to `soc/xtensa/nxp_adsp/Kconfig.defconfig` as it seems like all xtensa-based NXP SoCs use them.